packages/codemod/src/upgrade-legacy.ts

```diff
@@ -434,9 +431,12 @@ const upgradeLegacy = async () => {
             blitzClientProgram.toSource(),
           )
         } else {
-          throw new ExpectedError(
-            "Cookie Prefix is undefined & not a string. Please set your cookie prefix manually in app/blitz-client",
-          )
+          // Show error at end of codemod
+          collectedErrors.push({
+            message:
+              "Cookie Prefix is undefined & not a string. Please set your cookie prefix manually in app/blitz-client",
+            step: stepIndex!,
```

Comment:
Let's avoid unsafe type assertions as much as possible, especially in a new code. Since it can be undefined (as stated in `TStep` definition) but is required for the collectedErrors, can you either:

1) update the TStep type and make `stepIndex` required 
2) change this line to remove unsafe assertion (`!`) by providing a default value
3) modify `collectedErrors` type